web/src/elements/wizard/Wizard.ts

```diff
@@ -123,31 +184,72 @@ export class Wizard extends ModalButton {
         });
     }
 
+    /**
+     * Reset the wizard to it's initial state.
+     */
+    reset = () => {
+        this.open = false;
+
+        this.querySelectorAll("[data-wizardmanaged=true]").forEach((el) => {
+            el.remove();
+        });
+
+        for (const step of this.steps) {
+            const stepElement = this.getStepElementByName(step);
+
+            stepElement?.reset?.();
+        }
+
+        this.steps = this._initialSteps;
+        this.actions = [];
+        this.state = {};
+        this.activeStepElement = undefined;
+        this.canBack = true;
+        this.canCancel = true;
+    };
+
+    //#endregion
+
+    //#region Rendering
+
     renderModalInner(): TemplateResult {
-        const firstPage = this.querySelector<WizardPage>(`[slot=${this.steps[0]}]`);
-        if (!this.currentStep && firstPage) {
-            this.currentStep = firstPage;
+        const firstPage = this.getStepElementByIndex(0);
+
+        if (!this.activeStepElement && firstPage) {
+            this.activeStepElement = firstPage;
         }
-        const currentIndex = this.currentStep ? this.steps.indexOf(this.currentStep.slot) : 0;
-        let lastPage = currentIndex === this.steps.length - 1;
+
+        const activeStepIndex = this.activeStepElement
+            ? this.steps.indexOf(this.activeStepElement.slot)
+            : 0;
+
+        let lastPage = activeStepIndex === this.steps.length - 1;
+
         if (lastPage && !this.steps.includes("ak-wizard-page-action") && this.actions.length > 0) {
             this.steps = this.steps.concat("ak-wizard-page-action");
-            lastPage = currentIndex === this.steps.length - 1;
+            lastPage = activeStepIndex === this.steps.length - 1;
         }
+
+        const navigateToPreviousStep = () => {
+            const prevPage = this.getStepElementByIndex(activeStepIndex - 1);
+
+            if (prevPage) {
+                this.activeStepElement = prevPage;
+            }
+        };
+
         return html`<div class="pf-c-wizard">
             <div class="pf-c-wizard__header">
                 ${this.canCancel
                     ? html`<button
                           class="pf-c-button pf-m-plain pf-c-wizard__close"
                           type="button"
                           aria-label="${msg("Close")}"
-                          @click=${() => {
-                              this.reset();
-                          }}
+                          @click=${this.reset}
                       >
                           <i class="fas fa-times" aria-hidden="true"></i>
                       </button>`
-                    : html``}
+                    : nothing}
```

Comment:
AFAIK both `html`\`` and `nothing` appear interchangeable. Should we settle on one?